Flow.Launcher/MainWindow.xaml.cs

```diff
@@ -333,6 +333,28 @@ private void OnKeyDown(object sender, KeyEventArgs e)
                 _viewModel.SelectPrevPageCommand.Execute(null);
                 e.Handled = true;
             }
+            else if (e.Key == Key.Right)
+            {
+                int caretPosition = QueryTextBox.CaretIndex;
+                int queryLength = QueryTextBox.Text.Length;
+                if (caretPosition == queryLength && queryLength != 0)
```

Comment:
we should be able to use the viewmodel instead of relying too much on the control, i can update this logic here. @onesounds are you ok for me to make a commit to this PR with the change to use switch statement and viewmodel?

Reply:
sure, why not.

Reply:
i added the switch, and used the viewmodel's method to determine whether it's on the query results page or the context menu page.